networkx/readwrite/json_graph/tests/test_adjacency.py

```diff
@@ -4,13 +4,19 @@
 
 import networkx as nx
 from networkx.readwrite.json_graph import adjacency_data, adjacency_graph
+from networkx.utils import graphs_equal
 
 
 class TestAdjacency:
     def test_graph(self):
         G = nx.path_graph(4)
         H = adjacency_graph(adjacency_data(G))
-        nx.is_isomorphic(G, H)
+        assert nx.is_isomorphic(G, H)
+        assert graphs_equal(G, H)
+
+        H = adjacency_graph(json.loads(json.dumps(adjacency_data(G))))
```

Comment:
What does round-tripping through `json.loads`/`json.dumps` add here. Without the round tripping this is equivalent to the above test, and I don't think there's any added benefit to testing `json` serialization indirectly (that should be covered in the `json` package test suite).

If we're not testing something explicit here, I'd vote to remove these lines; if I'm missing something though I vote we add a comment explaining the test!
```suggestion
        H = adjacency_graph(json.loads(json.dumps(adjacency_data(G))))
```

Reply:
I added this to the test because test_cytoscape and test_node_link have tests that pass through dumps and loads, I'm not attached to it just wanted to ensure completeness between the three methods.

I presume it is to ensure that the graph_data is serialisable and deserialisable. I'm happy to remove or else add a comment explaining that it is that the graph_data produced is serialisable that is being tested (probably by moving to a new test method).

Reply:
Just found a case where

graphs_equal(G, H)  # True
graphs_equal(adjacency_graph(adjacency_data(G)))  # True
graphs_equal(adjacency_graph(json.loads(json.dumps(adjacency_data(G)))))  # False

Where non-String keys are String-ified and then not-unStringified during json.dumps.

So it may yet be a useful check

Reply:
Newest patchset now contains tests such that node_links and adjacency_data should be able to serialise non-String keys for the G.graph dict (so long as there's no nested dicts...).